src/custom/state/fee/updater.ts

```diff
@@ -17,8 +19,21 @@ function isDateLater(dateA: string, dateB: string): boolean {
 export default function FeesUpdater(): null {
   const { chainId } = useActiveWeb3React()
   const {
-    INPUT: { currencyId }
+    INPUT: { currencyId: sellToken },
+    OUTPUT: { currencyId: buyToken },
+    independentField,
+    typedValue: rawTypedValue
   } = useSwapState()
+  // debounce the typed value as to not overkill the useEffect
+  // fee API calculation/call
+  const typedValue = useDebounce(rawTypedValue, 200)
```

Comment:
as it recalculates on input change, we debounce here as to not overload system if user is typing

Reply:
yes, i second this. It feels a bit too fast doing request. I was getting requests while typing. Maybe 300 or 350 would be a good compromise. We increase the time 50-75% so I think it will decrease some request while you are typing, and still we don't add a delay on top of the normal query time...